support/sier-codec/src/schema.rs

```diff
@@ -243,4 +285,129 @@ mod tests {
             assert_eq!(struct_a.id(), struct_b.id());
         }
     }
+
+    #[cfg(test)]
+    mod parsing {
+        use super::*;
+
+        #[cfg(test)]
+        mod string {
+            use super::*;
+
+            #[test]
+            fn empty_string() {
+                let field = FieldDef {
+                    name: "foo".to_string(),
+                    type_: Type::String,
+                };
+
+                assert_eq!(
+                    field.parse(&[0]).unwrap(),
+                    (&[][..], Value::String("".to_string()))
+                );
+            }
+
+            #[test]
+            fn simple_string() {
+                let field = FieldDef {
+                    name: "foo".to_string(),
+                    type_: Type::String,
+                };
+
+                assert_eq!(
+                    field.parse(&[3, 65, 66, 67]).unwrap(),
+                    (&[][..], Value::String("ABC".to_string()))
+                );
+            }
+
+            #[test]
+            fn non_utf8() {
+                let field = FieldDef {
+                    name: "foo".to_string(),
+                    type_: Type::String,
+                };
+
+                assert!(field.parse(&[4, 0, 159, 146, 150]).is_err());
```

Comment:
Normally I don't really care about error types in applications, but since this is library code and we set a very specific error at https://github.com/trustfractal/protocol/pull/6/files#diff-d2c7017b3ce07f0414d04f03a7ec686c452d012307c09db316c9b176a7a4cb5dR57, can we assert that the error is the correct one and not a different type of error we might be causing?

For now, it's fairly obvious what the output would be but as the codebase grows, it might be helpful to keep track of that.

Reply:
Done.